packages/ripple-binary-codec/tools/generateDefinitions.js

```diff
@@ -0,0 +1,378 @@
+if (process.argv.length != 3 && process.argv.length != 4) {
+  console.error(
+    'Usage: ' +
+      process.argv[0] +
+      ' ' +
+      process.argv[1] +
+      ' path/to/rippled [path/to/pipe/to]',
+  )
+  process.exit(1)
+}
+
+////////////////////////////////////////////////////////////////////////
+//  Get all necessary files from rippled
+////////////////////////////////////////////////////////////////////////
+const path = require('path')
+
+const fs = require('fs')
+function readFile(filename) {
+  try {
+    return fs.readFileSync(filename).toString('utf-8')
+  } catch (err) {
+    console.error(`Error reading file ${filename}:`, err.message)
+    process.exit(1)
+  }
+}
+
+const sfieldHeaderFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/SField.h'),
+)
+const sfieldMacroFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/detail/sfields.macro'),
+)
+const ledgerFormatsMacroFile = readFile(
+  path.join(
+    process.argv[2],
+    '/include/xrpl/protocol/detail/ledger_entries.macro',
+  ),
+)
+const terFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/TER.h'),
+)
+const transactionsMacroFile = readFile(
+  path.join(
+    process.argv[2],
+    '/include/xrpl/protocol/detail/transactions.macro',
+  ),
+)
+
+const capitalizationExceptions = {
+  XCHAIN: 'XChain',
+}
+
+// Translate from rippled string format to what the binary codecs expect
+function translate(inp) {
+  try {
+    if (inp.match(/^UINT/m))
+      if (
+        inp.match(/256/m) ||
+        inp.match(/160/m) ||
+        inp.match(/128/m) ||
+        inp.match(/192/m)
+      )
+        return inp.replace('UINT', 'Hash')
+      else return inp.replace('UINT', 'UInt')
+
+    const nonstandardRenames = {
```

Comment:
I couldn't find any instances where `AMM`, `DIR_NODE` and `PAYCHAN` are used. The recent commits on `rippled:develop` do not trigger these keys in the dictionary.

Were they used in historical versions of the rippled codebase?

Reply:
https://github.com/XRPLF/rippled/blob/develop/include/xrpl/protocol/detail/ledger_entries.macro#L176